chia/wallet/vc_wallet/cr_cat_drivers.py

```diff
@@ -49,6 +49,49 @@
     package_or_requirement="chia.wallet.vc_wallet.cr_puzzles",
     include_standard_libraries=True,
 )
+CREDENTIAL_STRUCT: Program = Program.to(
+    (
+        (
+            (
+                SINGLETON_MOD_HASH,
+                SINGLETON_LAUNCHER_HASH,
+            ),
+            (
+                EXTIGENT_METADATA_LAYER_HASH,
+                EML_TP_COVENANT_ADAPTER_HASH,
+            ),
+        ),
+        (
+            Program.to(EXTIGENT_METADATA_LAYER_HASH)
+            .curry(
+                Program.to(EXTIGENT_METADATA_LAYER_HASH).get_tree_hash(),
+                Program.to(None),
+                GUARANTEED_NIL_TP,
+                GUARANTEED_NIL_TP.get_tree_hash(),
+                P2_ANNOUNCED_DELEGATED_PUZZLE,
+            )
+            .get_tree_hash_precalc(
```

Comment:
we have a function to compute the hash of a curried puzzle without actually produce the curried puzzle first, which is a lot faster. `curry_and_treehash()`.
I suppose this cost has been shifted to import-time, so it might not be as critical anymore. I would expect tests to benefit though, since they may import this module more times than the actually use this data structure.

Reply:
changed